components/exercises/chapter-5-oscar/Everything/WeatherApp.tsx

```diff
@@ -0,0 +1,162 @@
+              type='text'
+              value={locationInput}
+              placeholder='Enter location'
+              onChange={handleLocationInputChange}
+            />
+          </form>
+        </div>
+        {error && <p className='text-red-500'>{error}</p>}
+        <div className='flex'>
+          <div className='flex-1'>
+            <span className='mt-2 text-sm'>{weatherData.name}</span>
+            <div className='flex'>
+              <div className='flex-1'>
+                {weatherData.main && (
+                  <h1 className='text-4xl '>
+                    {weatherData.main.temp.toFixed()} ºC
+                  </h1>
+                )}
+              </div>
+              <div className='flex-1'>
+                {weatherData.main && (
+                  <p className='text-xs'>
+                    feels {weatherData.main.feels_like.toFixed()} ºC
+                  </p>
+                )}
+                {weatherData.main && (
+                  <p className='text-xs'>
+                    min {weatherData.main.temp_min.toFixed()} ºC
+                  </p>
+                )}
+                {weatherData.main && (
+                  <p className='text-xs'>
+                    max {weatherData.main.temp_max.toFixed()} ºC
+                  </p>
+                )}
+              </div>
+            </div>
+            <div>
+              {weatherData.weather && (
+                <p className='text-lg font-semibold'>
+                  {weatherData.weather[0].description}
+                </p>
+              )}
+            </div>
+          </div>
+          {weatherData.name != undefined && (
+            <div className='flex-1'>
+              <div className=' my-2 flex items-center justify-between gap-2 text-center'>
+                <div>
+                  {weatherData.main && (
+                    <p className='text-lg'>{weatherData.main.pressure}</p>
+                  )}
+                  <p className='text-sm'>Pressure</p>
+                </div>
+                <div>
+                  {weatherData.main && (
+                    <p className='text-lg'>{weatherData.main.humidity} %</p>
+                  )}
+                  <p className='text-sm'>Humidity</p>
+                </div>
+                <div>
+                  {weatherData.wind && (
+                    <p className='text-lg'>
+                      {weatherData.wind.speed.toFixed()} MPH
+                    </p>
+                  )}
+                  <p className='text-sm'>Wind</p>
+                </div>
+              </div>
+              {every(
+                [
+                  Number(weatherData.main?.temp.toFixed()) ?? 0,
+                  Number(weatherData.main?.temp_min.toFixed()) ?? 0,
+                  Number(weatherData.main?.feels_like.toFixed()) ?? 0,
+                ],
+                (n) => n <= (Number(weatherData.main?.temp_max.toFixed()) ?? 0)
```

Comment:
What does this do? This logic is a bit hard to follow, some comments or making it a function above with a descriptive name and calling it here would help

Reply:
Sorry, to make it clear I refactored it. It compares if the feel_like temperature is higher than the temp_max forecasted. Therefore, if it is higher, it will display "🔥 The temperature feels warmer than the maximum" else "👍 The temperature does not feel over the maximum"